src/blind.rs

```diff
@@ -456,4 +562,148 @@ impl Transaction {
         }
         Ok(())
     }
+
+    /// Blind a transaction
+    /// Blind all outputs but the fee outputs
+    /// As per the elements convention, In order to blind transaction, the user should set the blinding key
+    /// as the nonce field in the transaction.
+    /// If the nonce of the output is Null, it is not blinded
+    /// For a successful blind, atleast two outputs must be blinded.
+    pub fn blind<R, C>(
+        &mut self,
+        rng: &mut R,
+        secp: &Secp256k1<C>,
+        spent_utxo_secrets: &[(Asset, &TxOutSecrets)],
+    ) -> Result<Vec<(AssetBlindingFactor, ValueBlindingFactor)>, BlindError>
+    where
+        R: RngCore + CryptoRng,
+        C: Signing,
+    {
+        // Blinding Issuances unsupported
+        if self.input.iter().any(|i| i.has_issuance()) {
+            return Err(BlindError::IssuanceUnsupported)
+        }
+        // Everything must be explicit
+        if !self.output.iter().all(|o| o.asset.is_explicit() && o.value.is_explicit()) {
+            return Err(BlindError::MustHaveAllExplicitTxOuts)
+        }
+        // All outputs with script
+        let num_to_blind = self
+            .output
+            .iter()
+            .filter(|i| !i.is_fee() && i.nonce.is_confidential())
+            .count();
+        if num_to_blind < 2 {
```

Comment:
We should be able to blind a single output no problem.

Reply:
Fixed this and added a test.